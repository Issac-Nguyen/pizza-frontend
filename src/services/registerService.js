import axios from '../lib/axiosConfig';

function execute(value) {
    const api ='/register';
    return axios.post(api, value)
}

export default execute;