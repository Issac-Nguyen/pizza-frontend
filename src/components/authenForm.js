import React, {useState, useEffect} from 'react';
import {Row, Col, Tabs, Form, Input, Button, message, Spin} from 'antd';
import registerService from '../services/registerService';
import loginService from '../services/loginService'
import {connect} from 'react-redux'
import {closeAuthenForm, register, login} from '../actions/userActions';
import authLib from '../lib/auth';

const { TabPane } = Tabs;

const formItemLayout = {
    labelCol: {
      xs: { span: 24 },
      sm: { span: 8 },
    },
    wrapperCol: {
      xs: { span: 24 },
      sm: { span: 16 },
    },
  };
  const tailFormItemLayout = {
    wrapperCol: {
      xs: {
        span: 24,
        offset: 0,
      },
      sm: {
        span: 16,
        offset: 8,
      },
    },
  };

export const AuthenForm = (props, ref) => {
  let [sending, setSending] = useState(false);
  const [formRegiser] = Form.useForm();
  const [formLogin] = Form.useForm();
  
    const onFinishRegister = values => {
      setSending(true);
        const {email, password} = values;
        registerService({email, password}).then(res => {
          setSending(false);
          if(!res.data.success) {
            message.error(res.data.msg)
          } else {
            const token = res.data.data.token;
            if(token)
              authLib.saveToken(token)
            props.register(email);
            props.closeAuthenForm();
            formRegiser.resetFields();
          }
        }, err => {
          setSending(false);
        });
    };

    const onFinishLogin = values => {
      setSending(true);
        const {email, password} = values;
        loginService({email, password}).then(res => {
          setSending(false);
          if(!res.data.success) {
            message.error(res.data.msg)
          } else {
            const token = res.data.data.token;
            if(token)
              authLib.saveToken(token)
            props.login(email);
            props.closeAuthenForm();
            formLogin.resetFields();
          }
        }, err => {
          setSending(false);
        });
    }

    useEffect(() => {
      formRegiser.resetFields();
      formLogin.resetFields();
    }, [props.showAuthenForm])
    
    return (
    <Tabs defaultActiveKey="register">
        <TabPane tab="Register" key="register">
        <Form
        form={formRegiser}
            {...formItemLayout}
            name="register"
            onFinish={onFinishRegister}
            scrollToFirstError
            >
                <Form.Item
        name="email"
        label="E-mail"
        rules={[
          {
            type: 'email',
            message: 'The input is not valid E-mail!',
          },
          {
            required: true,
            message: 'Please input your E-mail!',
          },
          {
            max: 50,
            message: 'Only 50 characters'
          }
        ]}
      >
        <Input />
      </Form.Item>

      <Form.Item
        name="password"
        label="Password"
        rules={[
          {
            required: true,
            message: 'Please input your password!',
          },
          {
            whitespace: true,
            message: 'Password not valid'
          },
          {
            max: 16,
            message: 'Only 16 characters'
          }
        ]}
        hasFeedback
      >
        <Input.Password />
      </Form.Item>

      <Form.Item
        name="confirm"
        label="Confirm Password"
        dependencies={['password']}
        hasFeedback
        rules={[
          {
            required: true,
            message: 'Please confirm your password!',
          },
          ({ getFieldValue }) => ({
            validator(rule, value) {
              if (!value || getFieldValue('password') === value) {
                return Promise.resolve();
              }
              return Promise.reject('The two passwords that you entered do not match!');
            },
          }),
        ]}
      >
        <Input.Password />
      </Form.Item>
      <Row justify="center">
        <Spin spinning={sending} />
      </Row>
      <Form.Item {...tailFormItemLayout}>
        <Button type="primary" className="register-form-button" htmlType="submit" disabled={sending}>
          Register
        </Button>
      </Form.Item>
        </Form>
        </TabPane>
        <TabPane tab="Login" key="login">
        <Form
        form={formLogin}
        {...formItemLayout}
      name="normal_login"
      className="login-form"
      onFinish={onFinishLogin}
    >
      <Form.Item
        name="email"
        label="E-mail"
        rules={[
          {
            type: 'email',
            message: 'The input is not valid E-mail!',
          },
          {
            required: true,
            message: 'Please input your E-mail!',
          },
        ]}
      >
        <Input />
      </Form.Item>

      <Form.Item
        name="password"
        label="Password"
        rules={[
          {
            required: true,
            message: 'Please input your password!',
          },
        ]}
        hasFeedback
      >
        <Input.Password />
      </Form.Item>
      <Row justify="center">
        <Spin spinning={sending} />
      </Row>
      <Form.Item {...tailFormItemLayout}>
        <Button type="primary" htmlType="submit" className="login-form-button" disabled={sending}>
          Log in
        </Button>
      </Form.Item>
    </Form>
        </TabPane>
  </Tabs>
    )
}

const mapStateToProps = state => ({
  showAuthenForm: state.showAuthenForm,
});

const mapDispathToProps = dispatch => ({
  closeAuthenForm: () => {dispatch(closeAuthenForm())},
  register: (email) => {dispatch(register({email}))},
  login: (email) => {dispatch(login({email}))}
})

export default connect(mapStateToProps, mapDispathToProps)(AuthenForm);