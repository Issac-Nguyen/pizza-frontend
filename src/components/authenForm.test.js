import React from 'react'
import {render, cleanup, act, waitForElementToBeRemoved, getByText, getByRole,fireEvent, waitForDomChange, wait} from '@testing-library/react'
import {AuthenForm} from './authenForm'
import registerService from '../services/registerService'

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });

//   jest.mock('../services/registerService', () => () => {
//       return Promise.resolve({data: {
//         success: false,
//         msg: 'Email used',
//         testing: true
//     }})
//   })

  jest.mock('../services/registerService');

describe('authenForm', () => {
    afterEach(() => {
        cleanup();
        jest.clearAllMocks();
    })

    it('should take a snapshot', () => {
       const { asFragment } = render(<AuthenForm />)
       
       expect(asFragment(<AuthenForm />)).toMatchSnapshot()
    })
    it('should have register, login in document', () => {
        const { getByText, container } = render(<AuthenForm />)
        
        expect(container.querySelector('#tab-login')).toBeInTheDocument()
        expect(container.querySelector('#tab-register')).toBeInTheDocument()
     })

     it('should have snapshot login', () => {
        const { asFragment, container } = render(<AuthenForm />)
        
        fireEvent.click(container.querySelector('#tab-login'));
        expect(asFragment(<AuthenForm />)).toMatchSnapshot()
     })

     it('should not allow register wihout anything input', async () => {
        const { getByText, container } = render(<AuthenForm />)

        fireEvent.click(container.querySelector('button'));
        await waitForDomChange();
        expect(getByText(/please input your e-mail!/i)).toBeInTheDocument();
     })

     it('should call register service', async () => {
        const mockFn = registerService.mockResolvedValueOnce({
            data: {
                success: false,
                msg: 'Email used',
                testing: true
            }
        });
        const { getByText, container } = render(<AuthenForm />)
        fireEvent.change(container.querySelector('#register_email'), {targe: {value: 'abcddf@gmai.com'}})
        fireEvent.change(container.querySelector('#register_password'), {targe: {value: '1'}})
        fireEvent.change(container.querySelector('#register_confirm'), {targe: {value: '1'}})
        const formRegister = await container.querySelector('.register-form-button');
        fireEvent.submit(formRegister);
        // console.log(container.querySelector('#register'))
        // expect(mockFn).toHaveBeenCalledTimes(1);
        await wait();
        // expect(mockFn).toHaveBeenCalledTimes(1);
     });
})