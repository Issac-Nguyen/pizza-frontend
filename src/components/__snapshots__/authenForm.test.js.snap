// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authenForm should have snapshot login 1`] = `
<DocumentFragment>
  <div
    class="ant-tabs ant-tabs-top ant-tabs-line"
  >
    <div
      class="ant-tabs-bar ant-tabs-top-bar"
      role="tablist"
      tabindex="-1"
    >
      <div
        class="ant-tabs-nav-container"
      >
        <span
          class="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
          unselectable="unselectable"
        >
          <span
            class="ant-tabs-tab-prev-icon"
          >
            <span
              aria-label="left"
              class="anticon anticon-left ant-tabs-tab-prev-icon-target"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
          unselectable="unselectable"
        >
          <span
            class="ant-tabs-tab-next-icon"
          >
            <span
              aria-label="right"
              class="anticon anticon-right ant-tabs-tab-next-icon-target"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </span>
        </span>
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-scroll"
          >
            <div
              class="ant-tabs-nav ant-tabs-nav-animated"
            >
              <div>
                <div
                  aria-controls="tabpane-register"
                  aria-disabled="false"
                  aria-selected="false"
                  class=" ant-tabs-tab"
                  id="tab-register"
                  role="tab"
                  tabindex="-1"
                >
                  Register
                </div>
                <div
                  aria-controls="tabpane-login"
                  aria-disabled="false"
                  aria-selected="true"
                  class="ant-tabs-tab-active ant-tabs-tab"
                  id="tab-login"
                  role="tab"
                  tabindex="0"
                >
                  Login
                </div>
              </div>
              <div
                class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                style="transform: translate3d(0px,0,0); webkit-transform: translate3d(0px,0,0); width: 0px; display: block;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
      style="margin-left: -100%;"
    >
      <div
        aria-hidden="true"
        class="ant-tabs-tabpane ant-tabs-tabpane-inactive"
        role="tabpanel"
        style="visibility: hidden;"
        tabindex="-1"
      >
        <form
          class="ant-form ant-form-horizontal"
          id="register"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-required"
                for="register_email"
                title="E-mail"
              >
                E-mail
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="register_email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-required"
                for="register_password"
                title="Password"
              >
                Password
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-password ant-input-affix-wrapper"
                  >
                    <input
                      action="click"
                      class="ant-input"
                      id="register_password"
                      type="password"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        class="anticon anticon-eye-invisible ant-input-password-icon"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-required"
                for="register_confirm"
                title="Confirm Password"
              >
                Confirm Password
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-password ant-input-affix-wrapper"
                  >
                    <input
                      action="click"
                      class="ant-input"
                      id="register_confirm"
                      type="password"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        class="anticon anticon-eye-invisible ant-input-password-icon"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-row-center"
          >
            <div
              class="ant-spin"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-16 ant-col-sm-offset-8"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn register-form-button ant-btn-primary"
                    type="submit"
                  >
                    <span>
                      Register
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        aria-hidden="false"
        class="ant-tabs-tabpane ant-tabs-tabpane-active"
        role="tabpanel"
        style="visibility: visible;"
        tabindex="0"
      >
        <form
          class="ant-form ant-form-horizontal login-form"
          id="normal_login"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-required"
                for="normal_login_email"
                title="E-mail"
              >
                E-mail
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="normal_login_email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-required"
                for="normal_login_password"
                title="Password"
              >
                Password
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-password ant-input-affix-wrapper"
                  >
                    <input
                      action="click"
                      class="ant-input"
                      id="normal_login_password"
                      type="password"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        class="anticon anticon-eye-invisible ant-input-password-icon"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-row-center"
          >
            <div
              class="ant-spin"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-16 ant-col-sm-offset-8"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn login-form-button ant-btn-primary"
                    type="submit"
                  >
                    <span>
                      Log in
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`authenForm should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-tabs ant-tabs-top ant-tabs-line"
  >
    <div
      class="ant-tabs-bar ant-tabs-top-bar"
      role="tablist"
      tabindex="-1"
    >
      <div
        class="ant-tabs-nav-container"
      >
        <span
          class="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
          unselectable="unselectable"
        >
          <span
            class="ant-tabs-tab-prev-icon"
          >
            <span
              aria-label="left"
              class="anticon anticon-left ant-tabs-tab-prev-icon-target"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
          unselectable="unselectable"
        >
          <span
            class="ant-tabs-tab-next-icon"
          >
            <span
              aria-label="right"
              class="anticon anticon-right ant-tabs-tab-next-icon-target"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </span>
        </span>
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-scroll"
          >
            <div
              class="ant-tabs-nav ant-tabs-nav-animated"
            >
              <div>
                <div
                  aria-controls="tabpane-register"
                  aria-disabled="false"
                  aria-selected="true"
                  class="ant-tabs-tab-active ant-tabs-tab"
                  id="tab-register"
                  role="tab"
                  tabindex="0"
                >
                  Register
                </div>
                <div
                  aria-controls="tabpane-login"
                  aria-disabled="false"
                  aria-selected="false"
                  class=" ant-tabs-tab"
                  id="tab-login"
                  role="tab"
                  tabindex="-1"
                >
                  Login
                </div>
              </div>
              <div
                class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
      style="margin-left: 0%;"
    >
      <div
        aria-hidden="false"
        class="ant-tabs-tabpane ant-tabs-tabpane-active"
        role="tabpanel"
        style="visibility: visible;"
        tabindex="0"
      >
        <form
          class="ant-form ant-form-horizontal"
          id="register"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-required"
                for="register_email"
                title="E-mail"
              >
                E-mail
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input"
                    id="register_email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-required"
                for="register_password"
                title="Password"
              >
                Password
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-password ant-input-affix-wrapper"
                  >
                    <input
                      action="click"
                      class="ant-input"
                      id="register_password"
                      type="password"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        class="anticon anticon-eye-invisible ant-input-password-icon"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
            >
              <label
                class="ant-form-item-required"
                for="register_confirm"
                title="Confirm Password"
              >
                Confirm Password
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-password ant-input-affix-wrapper"
                  >
                    <input
                      action="click"
                      class="ant-input"
                      id="register_confirm"
                      type="password"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="eye-invisible"
                        class="anticon anticon-eye-invisible ant-input-password-icon"
                        role="img"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="eye-invisible"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                          />
                          <path
                            d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row ant-row-center"
          >
            <div
              class="ant-spin"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-16 ant-col-sm-offset-8"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <button
                    class="ant-btn register-form-button ant-btn-primary"
                    type="submit"
                  >
                    <span>
                      Register
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        aria-hidden="true"
        class="ant-tabs-tabpane ant-tabs-tabpane-inactive"
        role="tabpanel"
        style="visibility: hidden;"
        tabindex="-1"
      />
    </div>
  </div>
</DocumentFragment>
`;
